py/private/pytest_shard/pytest_shard.py

```diff
@@ -0,0 +1,73 @@
+"""
+This has been adapted from the original code at
+https://github.com/AdamGleave/pytest-shard
+The main difference is that the plugin has been converted to a class
+so that it can be passed to pytest.main's plugins attribute, and the
+shard selection was changed from a hash-based approach to a round-robin
+approach for a more balanced distribution with smaller numbers of tests.
+"""
+
+from typing import Iterable, List, Sequence
+
+from _pytest import nodes  # for type checking only
```

Comment:
Is it worth including type hints in this repo if we never run any type-checking? I'm genuinely not sure, maybe it's wise just for documentation, or in case we add type-checking later.

Reply:
it is nice to have for IDEs